features/device-config/persistence/impl/src/main/java/org/opennms/features/deviceconfig/persistence/impl/DeviceConfigDaoImpl.java

```diff
@@ -78,15 +78,15 @@ public DeviceConfigDaoImpl() {
     @Override
     public List<DeviceConfig> findConfigsForInterfaceSortedByDate(OnmsIpInterface ipInterface, String serviceName) {
 
-        return find("from DeviceConfig dc where dc.ipInterface.id = ? AND serviceName = ? ORDER BY lastUpdated DESC",
+        return find("from DeviceConfig dc where dc.lastUpdated is not null AND dc.ipInterface.id = ? AND serviceName = ? ORDER BY lastUpdated DESC",
```

Comment:
May be need to provide an option to query all DeviceConfigs in one of the DAO calls.

Reply:
This can currently be done via `findMatching(criteria)` from parent `AbstractDaoHibernate` class. Currently used in `DefaultDeviceConfigRestService.getDeviceConfigs`.

In separate PR, we can add the `status` as a field to sort/filter on

Reply:
okay, we can update later!